src/components/utils/index.js

```diff
@@ -29,6 +29,18 @@ let isIOS = (/iPad|iPhone|iPod/.test(navigator.platform || '') || (navigator.pla
 // Copied from https://github.com/muaz-khan/DetectRTC/blob/master/DetectRTC.js
 export const isDesktop = !(/Android|webOS|BB10|BlackBerry|IEMobile|Opera Mini|Mobile|mobile/i.test(navigator.userAgent || '')) && !isIOS
 
+export let hasEnvironmentCamera;
+
+navigator.mediaDevices.getUserMedia(
```

Comment:
On browsers like Safari or Firefox that do not remember media permissions by default, this will cause the user to be prompted to approve media permissions multiple times. Additionally, if I remember correctly some browsers do not support the `{facingMode: { exact: ... }}` syntax so this might create a webcam crash. Is there any way to determine if the device has environment camera or is hybrid?

Reply:
The only browser that I noticed requested permission an annoying amount was firefox mobile. Safari only seems to request it when you first navigate  to the site. Firefox for desktop can be annoying, but if you check the remember permission box the first time, it won't ask again. As far as the browser support for facingMode, this check is just to set the boolean and will catch any exceptions, the actual constraint is passed through the prop into the webcam element in DocumentLiveCapture.js. I did notice that desktop Safari would ignore the facingMode and give the front camera if `useLiveDocumentCapture=true`. 

I don't know how much we want to consolidate device, user media, and browser detection, but if we only just want to identify Surface tablets, I think we could do something like `isDesktop&&navigator.platform==='Win32'&&<the facing mode check>` Firefox and Edge both support the facingMode constraint, Chrome doesn't however, which is why I have a workaround in my PR for the react-webcam repo that enumerates devices and checks the labels for the words 'back' or 'rear'.

Reply:
> Safari only seems to request it when you first navigate to the site

This depends on the Mac OS and user settings. If I remember correctly, from Mojave onwards, the user needs to allow camera permissions each time.

Reply:
Wouldn't it be possible to handle this inside react-webcam?

Reply:
> Wouldn't it be possible to handle this inside react-webcam?

If we don't detect that the device is a Surface in the JS SDK, we don't have a way to direct it to use the DocumentLiveCapture component, because it registers as a desktop device.